pkg/service/model.go

```diff
@@ -41,6 +82,64 @@ func (s *Service) ConvertToMDModel(ctx context.Context, fileUID uuid.UUID, calle
 	// Get the appropriate prefix for the file type
 	prefix := getFileTypePrefix(fileType)
 
+	// If PDF, split into batches and process in parallel
+	if fileType == artifactpb.FileType_FILE_TYPE_PDF {
+		pdfData, err := base64.StdEncoding.DecodeString(fileBase64)
+		if err != nil {
+			logger.Error("Failed to decode base64 PDF", zap.Error(err))
+			return "", err
+		}
+		batches, err := splitPDFIntoBatches(pdfData, 5)
+		if err != nil {
+			logger.Error("Failed to split PDF into batches", zap.Error(err))
+			return "", err
+		}
+		results := make([]string, len(batches))
+		errCh := make(chan error, len(batches))
+		var wg sync.WaitGroup
+		for i, b64 := range batches {
+			wg.Add(1)
+			go func(idx int, b64 string) {
+				defer wg.Done()
```

Comment:
nit: part of the code in this function is duplicated from the non-pdf flow. A better approach would be treating every file the same, allowing batching by default. For PDFS:

```go
var filesToProcess []string
switch fileType {
case artifactpb.FileType_FILE_TYPE_PDF:
	// ... 
	batches, err := splitPDFIntoBatches(pdfData, 5)
	if err != nil { /* ... */ }
	filesToProcess = batches
default:
	filesToProcess = []string{fileBase64}
}

// ...

for i, b64 := range filesToProcess { /* ... */ }	
```